src/pages/managerForm.js

```diff
@@ -1,66 +1,209 @@
-import React from 'react';
-import Link from 'next/link';
-import Image from 'next/image'
-import NavBar from '@/components/Header/NavBar';
-import CustomizableButton from '@/components/common/CustomizableButton';
-import FooterBar from '@/components/common/FooterBar';
-import styles from '../styles/managerForm.module.css';
+import { useState, useRef } from "react";
+import React from "react";
+import { useRouter } from "next/navigation";
+import Link from "next/link";
+import Image from "next/image";
+import NavBar from "@/components/Header/NavBar";
+import CustomizableButton from "@/components/common/CustomizableButton";
+import FooterBar from "@/components/common/FooterBar";
+import { imageUpload, uploadFile, deleteFile, resetInput, goHome } from "@/pages/helperFile";
+import styles from "../styles/managerForm.module.css";
+import { firebaseDb, firebaseStorage } from "@/firebase/config";
+import {collection, addDoc} from "firebase/firestore";
+import { uploadBytes, getDownloadURL, ref } from "firebase/storage";
 
 
-const managerForm = () => {
-    
-    const categoryChoice = () => {
-        console.log("clicked the button")
+function ManagerForm() {
+  const initialFormData = {
+    community_name: "",
+    community_manager: "",
+    community_location: "",
+    manager_url: "",
+    manager_linkedin: "",
+    manager_twitter: "",
+    community_date: "",
+    community_vision: "",
+    community_description: "",
+  };
+  const [formData, setFormData] = useState(initialFormData);
+  const fileInputRef = useRef(null);
+  const formRef = useRef(null);
+  const [fileUpload, setFileUpload] = useState();
+  const router = useRouter();
+  const handleChange = (event) => {
+    event.preventDefault();
+    setFormData({ ...formData, [event.target.name]: event.target.value });
+  };
+  const inputFields = [
```

Comment:
I think this can be extracted to a ManagerFormHelper file. Create a function that returns the appropriate fields. This will make it easier to test. Then you can put other manager form related functions.